api/v1/menus/create.js

```diff
@@ -30,10 +31,21 @@ module.exports = (req, res, next) => {
     fsClient.uploadMenu(req.file)
         .then(id => {
             menuEntry.file = `/files/${id}`
-            return deleteFile(req.file.path);
+            return;
         })
         .then(() => dbClient.insert(menuEntry))
-        .then(dbEntry => res.status(httpStatus.OK_CREATED).send(dbEntry))
+        .then(dbEntry => {
+            // While is converting the PDF to text it finishes with ocr key like that: []
+            let ocrInstance = new OCR();
+            ocrInstance.getText(req.file.path).then((data) => {
+                dbEntry.ocr.push(data);
+                return deleteFile(req.file.path);
+            }).then(() => dbClient.update(dbEntry))
+                .catch((err) => {
+                    console.log(err);
+                })
+            return res.status(httpStatus.OK_CREATED).send(dbEntry); 
+        })
```

Comment:
We can better use Promises features here, even thought Promises are being used, there's still a bunch of callbacks here and there, I built the following pipeline (not tested):

```js
fsClient.uploadMenu(req.file)
.then(id => {
    menuEntry.file = `/files/${id}`
    return;
})
.then(() => {
    let ocrInstance = new OCR();
    return ocrInstance.getText(req.file.path)
})
.then((data) => {
    menuEntry.ocr.push(data);
    return deleteFile(req.file.path);
})
.then(() => dbClient.insert(menuEntry))
.then((dbEntry) => {
    return res.status(httpStatus.OK_CREATED).send(dbEntry); 
})
```

Reply:
When using promises we must always avoid nested promises (handling promises inside promises), because this makes code readability worse, and sometimes may lead to unintended behavior (there's more routes to account for). Also, the intention of using promises is to not have something similar to callbacks chain.

Reply:
@gabrieldrs I think, for user experience we must return the document on the DB without the ocr array fulfilled... because the tesseract spends much time processing the file... Let me know your opinion.